src/tests/dictionaryConcepts/container/EditConcept.test.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import Router from 'react-mock-router';
+import {
+  EditConcept,
+  mapStateToProps,
+} from '../../../components/dictionaryConcepts/containers/EditConcept';
+import { newConcept, existingConcept } from '../../__mocks__/concepts';
+
+jest.mock('uuid/v4', () => jest.fn(() => 1234));
+jest.mock('react-notify-toast');
+
+global.localStorage = {
+  getItem: () => '/',
+};
+
+describe('Test suite for dictionary concepts components', () => {
+  const props = {
+    match: {
+      params: {
+        conceptType: 'diagnosis',
+        collectionName: 'dev-col',
+        type: 'users',
+        typeName: 'emmabaye',
+        language: 'en',
+      },
+    },
+    history: {
+      push: jest.fn(),
+    },
+    createNewName: jest.fn(),
+    addNewDescription: jest.fn(),
+    clearSelections: jest.fn(),
+    fetchExistingConcept: jest.fn(),
+    clearPreviousConcept: jest.fn(),
+    createNewNameForEditConcept: jest.fn(),
+    removeDescriptionForEditConcept: jest.fn(),
+    addDescriptionForEditConcept: jest.fn(),
+    removeNameForEditConcept: jest.fn(),
+    updateConcept: jest.fn(),
+    newName: ['1'],
+    description: ['1'],
+    existingConcept: {
+      names: [{
+        uuid: '1234',
+      }],
+      descriptions: [{
+        uuid: '1234',
+      }],
+    },
+  };
+  it('should render without breaking', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} />
+    </Router>);
+    expect(wrapper.find('h3').text()).toEqual(': Edit a diagnosis Concept ');
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should handle form completion and submission with invalid/incomplete data', () => {
```

Comment:
@emmabaye , what is going on in this test seems a bit different from the description. I see there is a `remove description click` event. I think that should be a different test on its own. And it would be better to test for functionality of the application rather than testing for the snapshot matching. I think match snapshot should be used to test whether something rendered correctly. Not when you are handling events.

What happens when you change, or delete, expect that. That is just my opinion. What do you think?

Reply:
There were other tests besides snapshot tests that simulated events

Reply:
Please refactor this test to split concerns. Let the remove on click be on its own. Because adding it in here declares the onClick event handler tested yet you are not asserting anything as far as it's concerned. Otherwise good stuff.

Reply:
Thanks for the feedback. I will do that.

Reply:
Updated